File: app/src/main/java/zyon/notifier/notification/NotiAdapter.java
package zyon.notifier.notification;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.lang.ref.WeakReference;
import java.util.ArrayList;

import zyon.notifier.MainActivity;
import zyon.notifier.R;

public class NotiAdapter extends RecyclerView.Adapter<NotiAdapter.CustomViewHolder> {

    private ArrayList<Notification> mList;
    WeakReference<Context> mContextWeakReference;

    public class CustomViewHolder extends RecyclerView.ViewHolder {

        protected View color;
        protected TextView title;
        protected TextView text;
        protected LinearLayout parent;

        public CustomViewHolder(View view, final Context context) {
            super(view);

            this.color = view.findViewById(R.id.list_main_color);
            this.title = view.findViewById(R.id.list_main_title);
            this.text = view.findViewById(R.id.list_main_text);
            this.parent = view.findViewById(R.id.list_main);

            parent.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    ((MainActivity) context).Edit( getAdapterPosition() );
                }
            });

        }

    }

    public NotiAdapter(ArrayList<Notification> list, Context context) {

        this.mList = list;
        this.mContextWeakReference = new WeakReference<Context>(context);

    }

    @Override
    public CustomViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {

        Context context = mContextWeakReference.get();
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.list_main, viewGroup, false);
        CustomViewHolder viewHolder = new CustomViewHolder(view, context);

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull CustomViewHolder viewholder, int position) {

        viewholder.color.setBackgroundColor( Color.parseColor( mList.get(position).getColor() ) );
        viewholder.title.setText(mList.get(position).getTitle());
        viewholder.text.setText(mList.get(position).getText());

    }

    @Override
    public int getItemCount() { return (null != mList ? mList.size() : 0); }

}

File: app/src/main/java/zyon/notifier/notification/NotiAdapter.kt
package zyon.notifier.notification

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import java.util.*
import java.lang.ref.WeakReference

import zyon.notifier.MainActivity
import zyon.notifier.R

class NotiAdapter(private val mList: ArrayList<Notification>?, context: Context) : RecyclerView.Adapter<NotiAdapter.CustomViewHolder>() {

    private val mContextWeakReference = WeakReference(context)

    inner class CustomViewHolder(view: View, context: Context?) : RecyclerView.ViewHolder(view) {

        private var parent: LinearLayout = view.findViewById(R.id.list_main)
        val color: View = view.findViewById(R.id.list_main_color)
        val title: TextView = view.findViewById(R.id.list_main_title)
        val text: TextView = view.findViewById(R.id.list_main_text)

        init {

            // click anywhere
            parent.setOnClickListener {

                (context as MainActivity?)!!.Edit(adapterPosition)

            }

        }

    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): CustomViewHolder {

        val context = mContextWeakReference.get()
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.list_main, viewGroup, false)

        return CustomViewHolder(view, context)

    }

    override fun onBindViewHolder(viewholder: CustomViewHolder, position: Int) {

        viewholder.color.setBackgroundColor(Color.parseColor(mList!![position].color))
        viewholder.title.text = mList[position].title
        viewholder.text.text = mList[position].text

    }

    override fun getItemCount(): Int { return mList?.size ?: 0 }

}
